jax/interpreters/pxla.py

```diff
@@ -196,6 +196,7 @@ def shard_args(devices: Sequence[xb.xla_client.Device],
         buffers[r][a] = (buf if buf.device() == devices[r]
                          else buf.copy_to_device(devices[r]))
     else:
+      arg = xla.canonicalize_dtype(arg)
```

Comment:
Why did you need this? Maybe we should just overload the necessary handler more? (There's a TODO to get rid of canonicalize_dtype but I don't know anything about it)

Reply:
With this change, users can now end up passing scalars to pmap; that wasn't handled before because it couldn't happen at top level. You're right that we could add a scalar handler instead; I'm following xla.py here but I didn't realize we wanted to kill `canonicalize_dtype`. Think I should switch to a handler?

(Similarly, the change to `spec_to_indices` is because we are now calling that on specs that have scalar avals).

Reply:
Ah makes sense. I would say keep it as-is, I'm not totally sure what the TODO is proposing anyway so may as well be consistent for now.